src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -151,20 +166,17 @@ export class DefaultTelemetryService implements TelemetryService {
                 this.publisher.enqueue(...this._eventQueue)
                 await this.publisher.flush()
                 this.clearRecords()
+                this._listeners.forEach(l => l())
             }
         }
     }
 
-    // TODO: replace this with `setInterval`
-    private async startTimer(): Promise<void> {
-        this._timer = globals.clock.setTimeout(
-            // this is async so that we don't have pseudo-concurrent invocations of the callback
-            async () => {
-                await this.flushRecords()
-                this._timer?.refresh()
-            },
-            this._flushPeriod
-        )
+    private startTimer(): void {
+        // The timer is not reset until after the flush as completed
+        this._timer = globals.clock.setTimeout(async () => {
+            await this.flushRecords()
```

Comment:
Fake timers do not await callbacks, so need to fix this at some point.

Reply:
Fake timers continue to be a burden. Perhaps it makes sense to make the timer values globally-controllable, so that tests can control them. Then real timers are used, but with adjusted times.

Reply:
Yup agreed. I'm thinking of ways to move away from it. The problem with real timers is that they are very inaccurate. So test code would need to account for this :/